src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/TestStatus.java

```diff
@@ -0,0 +1,56 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.util.ObjectUtils;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example: STS++CO'
+ */
+@Getter
+@Builder
+@RequiredArgsConstructor
+public class TestStatus extends Segment {
+
+    public static final String KEY = "STS";
+
+    @NonNull
+    private final TestStatusCode testStatusCode;
+
+    public static TestStatus fromString(String edifactString) {
+        if (!edifactString.startsWith(KEY)) {
+            throw new IllegalArgumentException("Can't create " + TestStatus.class.getSimpleName() + " from " + edifactString);
+        }
+
+        String[] keySplit = Split.byPlus(edifactString);
+        String code = keySplit[2];
+
+        return new TestStatus(TestStatusCode.fromCode(code));
```

Comment:
Is it how it should be?

<img width="1016" alt="Screenshot 2021-02-04 at 16 19 29" src="https://user-images.githubusercontent.com/36967438/106922281-cc4c1380-6704-11eb-803d-73617ba8f994.png">

Reply:
Works fine for me - looks like IntelliJ hasn't picked up lombok's woven constructor; double check annotation processing is enabled, and as a last resort you can use File > invalidate caches/restart

Reply:
Strange, I don't have that error. Try doing `File` -> `Invalidate Caches / Restart...`, and if that doesn't work, try deleting the branch locally, and pull it down again, and see if the error's still there